src/Configuration/test/CloudFoundry.Test/CloudfoundryConfigurationProviderTest.cs

```diff
@@ -216,6 +224,96 @@ public void Load_VCAP_APPLICATION_Allows_Reload_Without_Throwing_Exception()
+            options.ForwardedHeaders.Should().HaveFlag(ForwardedHeaders.XForwardedProto);
+            options.KnownNetworks.Should().BeEmpty();
+            options.KnownProxies.Should().BeEmpty();
+        }
+        else
+        {
+            options.ForwardedHeaders.Should().Be(ForwardedHeaders.None);
+            options.KnownNetworks.Should().ContainSingle().Which.Should().BeEquivalentTo(IPNetwork.Parse("127.0.0.1/8"));
+            options.KnownProxies.Should().ContainSingle().Which.Should().Be(IPAddress.Parse("::1"));
+        }
+    }
+
+    [Fact]
+    public async Task ForwardedHeadersOptions_can_be_customized_when_running_on_CloudFoundry()
+    {
+        using var vcapScope = new EnvironmentVariableScope("VCAP_APPLICATION", "{}");
+        WebApplicationBuilder builder = TestWebApplicationBuilderFactory.CreateDefault();
+        builder.AddCloudFoundryConfiguration();
+        builder.Services.Configure<ForwardedHeadersOptions>(options => options.KnownProxies.Add(IPAddress.Parse("192.168.1.20")));
+        await using WebApplication host = builder.Build();
+        bool? forwardedHeadersWereEvaluated = null;
+
+        host.Map("/", context =>
+        {
+            forwardedHeadersWereEvaluated = context.Request.IsHttps;
+            return Task.CompletedTask;
+        });
+
+        await host.StartAsync(TestContext.Current.CancellationToken);
+        TestServer server = host.GetTestServer();
+
+        HttpContext responseContext = await server.SendAsync(requestContext =>
+        {
+            requestContext.Request.Headers["X-Forwarded-Proto"] = "https";
+            requestContext.Request.Headers["X-Forwarded-For"] = "192.168.1.19";
+            requestContext.Connection.RemoteIpAddress = IPAddress.Loopback;
+        }, TestContext.Current.CancellationToken);
+
+        responseContext.Response.StatusCode.Should().Be(200);
+        forwardedHeadersWereEvaluated.Should().BeFalse("X-Forwarded-Proto should not be evaluated for unknown proxies");
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task ForwardedHeadersMiddleware_updates_connection_details_when_running_on_CloudFoundry(bool isRunningOnCloudFoundry)
+    {
+        using var vcapScope = new EnvironmentVariableScope("VCAP_APPLICATION", isRunningOnCloudFoundry ? "{}" : null);
+        WebApplicationBuilder builder = TestWebApplicationBuilderFactory.CreateDefault();
+        builder.AddCloudFoundryConfiguration();
+        await using WebApplication host = builder.Build();
+        bool? forwardedHeadersWereEvaluated = null;
+
+        host.Map("/", context =>
+        {
+            forwardedHeadersWereEvaluated = context.Request.IsHttps;
+            return Task.CompletedTask;
+        });
+
+        await host.StartAsync(TestContext.Current.CancellationToken);
+        TestServer server = host.GetTestServer();
+
+        HttpContext responseContext = await server.SendAsync(requestContext =>
```

Comment:
I don't understand the names here. An `HttpContext` contains both request and response data. Why not name both of them `httpContext`?